streams/src/main/java/org/apache/kafka/streams/kstream/internals/KTableImpl.java

```diff
@@ -78,14 +78,24 @@
 
     private final Serde<K> keySerde;
     private final Serde<V> valSerde;
+    private final String storeName;
 
     private boolean sendOldValues = false;
 
+
     public KTableImpl(KStreamBuilder topology,
                       String name,
                       ProcessorSupplier<?, ?> processorSupplier,
                       Set<String> sourceNodes) {
-        this(topology, name, processorSupplier, sourceNodes, null, null);
+        this(topology, name, processorSupplier, sourceNodes, null, null, null);
+    }
+
+    public KTableImpl(KStreamBuilder topology,
```

Comment:
@enothereska I'd prefer to do https://issues.apache.org/jira/browse/KAFKA-3911 in a separate JIRA because:

1. Keeping one feature / bug-fix in one PR unless they are really trivial makes reviewing easier; especially for this patch that is already dragging long with large number of class files changed.

2. KAFKA-3911 itself is complex enough, for example we should think whether to remove the `enableSendingOldValues` in `KTableProcessorSupplier` since with this fix we are ALWAYS sending old values. This ticket requires thinking through a bit more.